### PR TITLE
Toc updates for accessibility, part 1

### PR DESCRIPTION
**Pull request by BillWagner**
Update a variety of TOC nodes to remove parent nodes with both content and children.

This is trying an experiment in the framework to see if it provides a better navigation experience.

### CODE REVIEW

**Comment by BillWagner** [comment]
This ended up bigger than I thought it would be. (The LINQ section had a lot more updates than I expected).  Here's a summary of the kinds of changes made:

1. Many of the "parent" toc nodes with children were portal pages, without much usable content. Those were removed and the content redirected to the first child, or occasionally the parent.
1. Some parent toc nodes provides a good overview or introductory topic to the section. In those cases, the TOC was updated to add a new TOC text and make the node a child node.
1. Some child sections had a small set of nodes. Those children were promoted up one level.

To clean links, when a child node had a single "see also" link to the parent, that link was removed.

**Review by mairaw — COMMENTED** [review]
I've reviewed up to advanced-linq-to-xml-programming.md. Looking good so far @BillWagner. Submitting my comments so far while I grab something to eat and then I'll go back to this.

File: docs/csharp/language-reference/keywords/let-clause.md [hunk 1]
```diff
@@ -30,5 +30,5 @@ In the following example `let` is used in two ways:
 - [C# Reference](../../language-reference/index.md)
 - [Query Keywords (LINQ)](query-keywords.md)
 - [Language Integrated Query (LINQ)](../../linq/index.md)
-- [Getting Started with LINQ in C#](../../programming-guide/concepts/linq/getting-started-with-linq.md)
+- [Getting Started with LINQ in C#](../../programming-guide/concepts/linq/index.md)
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/keywords/let-clause.md]: Yes. The LINQ section really needs updates. That's in the long-term plan.

File: .openpublishing.redirection.json [hunk 2]
```diff
@@ -414,6 +426,42 @@
...
+            "source_path": "docs/csharp/language-reference/keywords/literal-keywords.md",
+            "redirect_url": "/dotnet/csharp/language-reference/keywords"
+        },
+        {
+            "source_path": "docs/csharp/language-reference/keywords/modifiers.md",
+            "redirect_url": "/dotnet/csharp/language-reference/keywords"
+        },
+        {
+            "source_path": "docs/csharp/language-reference/keywords/namespace-keywords.md",
+            "redirect_url": "/dotnet/csharp/language-reference/keywords/namespace"
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply .openpublishing.redirection.json]: I think I should change this.  Good catch.

My original thinking was that it was likely that readers were trying to find "namespace". But, the removed page had several keywords, so the index page would be better. Updated.

File: docs/csharp/programming-guide/concepts/assemblies-gac/toc.yml [hunk 3]
```diff
@@ -2,7 +2,6 @@
   href: index.md
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/programming-guide/concepts/assemblies-gac/toc.yml]: Wow. what a catch.  All of the articles in this directory have been redirected to the standard folder.  I checked the stats, and *none* of them have any hits in the last 30 days. I've updated the PR to remove these nodes, and remove those articles. (They had already been deleted).

File: docs/csharp/programming-guide/concepts/covariance-contravariance/toc.yml [hunk 4]
```diff
@@ -3,15 +3,13 @@
   items:
   - name: Variance in Generic Interfaces
     href: variance-in-generic-interfaces.md
-    items:
-    - name: Creating Variant Generic Interfaces
-      href: creating-variant-generic-interfaces.md
-    - name: Using Variance in Interfaces for Generic Collections
-      href: using-variance-in-interfaces-for-generic-collections.md
+  - name: Creating Variant Generic Interfaces
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/programming-guide/concepts/covariance-contravariance/toc.yml]: I agree. I looked at it a few times, and that's a better choice.

**Comment by BillWagner** [comment]
This was broken into several PRs, and each of those has been merged.  Closing.

**Pull request closed without merging**